pyo3-bytes/src/bytes.rs

```diff
@@ -62,6 +63,66 @@ impl PyBytes {
     pub fn as_slice(&self) -> &[u8] {
         self.as_ref()
     }
+
+    /// Slice the underlying buffer using a Python slice object
+    ///
+    /// This should behave the same as Python's byte slicing:
+    ///     - `ValueError` if step is zero
+    ///     - Negative indices a-ok
+    ///     - If start/stop are out of bounds, they are clipped to the bounds of the buffer
+    ///     - If start > stop, the slice is empty
+    ///
+    /// This is NOT exposed to Python under the `#[pymethods]` impl
+    fn slice(&self, slice: &Bound<'_, PySlice>) -> PyResult<PyBytes> {
+        let len_isize = self.0.len() as isize;
+        let psi = slice.indices(len_isize)?;
+        let (start, stop, step) = (psi.start, psi.stop, psi.step);
+        if step == 0 {
+            return Err(PyValueError::new_err("step is zero"));
+        }
+
+        // I think this is right!?
```

Comment:
Can we have a test to test this case specifically? I don't know what it's doing

Reply:
Which one? step=0? I think there is a test for that.